testsuite/MDAnalysisTests/coordinates/test_lammps.py

```diff
@@ -507,3 +509,86 @@ def test_atom_reordering(self, u, reference_positions):
                                              reference_positions['atom13_pos']):
             assert_almost_equal(atom1.position, atom1_pos, decimal=5)
             assert_almost_equal(atom13.position, atom13_pos, decimal=5)
+
+
+@pytest.mark.parametrize("convention",["unscaled","unwrapped","scaled_unwrapped"])
+def test_open_absent_convention_fails(convention):
+    with pytest.raises(ValueError, match="no coordinate information of type"):
+        mda.Universe(LAMMPSDUMP, format='LAMMPSDUMP',
+                       lammps_coordinate_convention=convention)
+
+def test_open_incorrect_convention_fails():
+    with pytest.raises(ValueError, match="coordinate convention incorrectly specified"):
+        mda.Universe(LAMMPSDUMP, format='LAMMPSDUMP',
+                       lammps_coordinate_convention="42")
+
+
+@pytest.mark.parametrize("convention",["auto", "unscaled", "scaled",
+                         "unwrapped","scaled_unwrapped"])
+def test_open_all_convention(convention):
+    mda.Universe(LAMMPSDUMP_allcoords, format='LAMMPSDUMP',
+                       lammps_coordinate_convention=convention)
+
+class TestCoordinateMatches(object):
+    @pytest.fixture()
+    def universes(self):
+        coordinate_conventions = ["unscaled", "scaled", "unwrapped",
+                    "scaled_unwrapped"]
+        universes = {i: mda.Universe(LAMMPSDUMP_allcoords, format='LAMMPSDUMP',
+                       lammps_coordinate_convention=i) for i in coordinate_conventions}
+        return universes
+    
+    @pytest.fixture()
+    def reference_unscaled_positions(self):
+        # copied from trajectory file
+        # atom 340 is the first one in the trajectory so we use that
+        atom340_pos1_unscaled = [4.48355, 0.331422, 1.59231]
+        atom340_pos2_unscaled = [4.41947, 35.4403, 2.25115]
+        atom340_pos3_unscaled = [4.48989, 0.360633, 2.63623]
+        return np.asarray([atom340_pos1_unscaled, atom340_pos2_unscaled,atom340_pos3_unscaled])
+    
+    def test_unscaled_reference(self, universes, reference_unscaled_positions):
+        atom_340 = universes["unscaled"].atoms[339]
+        for i,ts_u in enumerate(universes["unscaled"].trajectory[0:3]):
+            assert_almost_equal(atom_340.position, reference_unscaled_positions[i,:], decimal=5)
+    
+    def test_scaled_reference(self, universes, reference_unscaled_positions):
```

Comment:
@richardjgowers @IAlibay question surrounding the accuracy of the `StoR` transform from `lib.distances`. Is this function known to be ill conditioned or something? This only seems to match to 1 dp.

Reply:
Now a separate issue #3361

Reply:
1 d.p. sounds problematic, you kinda have to try to get that bad a float match 😕 
Was this the same issue with the old StoR transformation or is it now only becoming an issue with this code?

Reply:
Yeah no changes made to `StoR()` here, I have raised an issue #3361, as will need to validate on a toy system.